src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_SIZE = 25;
+    private static final int DEFAULT_RADIUS = 10;
+    private static final int TOTAL_NUMBER_OF_FIGURES = 5;
+
+    private Figure[] randomFigure = new Figure[TOTAL_NUMBER_OF_FIGURES];
+    private Random random = new Random();
+
+    public int getRandom() {
+        return random.nextInt(MAX_SIZE);
+    }
+
+    public void setRandom(Random random) {
+        this.random = random;
+    }
+
+    public void setRandomFigure(Figure[] randomFigure) {
+        this.randomFigure = randomFigure;
+    }
+
+    public Figure getRandomFigure() {
+        randomFigure[0] = new IsoscelesTrapezoid(getRandom(), getRandom(), getRandom());
+        randomFigure[1] = new RightTriangle(getRandom(), getRandom());
+        randomFigure[2] = new Rectangle(getRandom(), getRandom());
+        randomFigure[3] = new Circle(getRandom());
+        randomFigure[4] = new Square(getRandom());
+        int index = new Random().nextInt(randomFigure.length);
+        return randomFigure[index];

```

Comment:
dont create all the figures, add switch case to create only 1 figure each time you invoke the method
add method without parameters to get random figure of each type
getRandomCircle()
getRandomSquare()